storm-server/src/main/java/org/apache/storm/blobstore/BlobStoreUtils.java

```diff
@@ -191,6 +192,8 @@ public static boolean downloadUpdatedBlob(Map<String, Object> conf, BlobStore bl
                     out.close();
                 }
                 isSuccess = true;
+            } catch(FileNotFoundException fnf) {
```

Comment:
I'd rather leave it as it is, unless we feel confident that it is OK to swallow here.

Reply:
I agree with @HeartSaVioR here.

Reply:
Btw, theres a race condition where the topology gets killed while the blob download is going on and apparently it throws a FNF exception here when invoking out.close().